ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/PlatformApplications.scala

```diff
@@ -72,6 +74,23 @@ object PlatformApplications {
     def withMaxNumberOfAcsContracts(cap: Int) = copy(maxNumberOfAcsContracts = Some(cap))
 
     def withCommandConfiguration(cc: CommandConfiguration) = copy(commandConfiguration = cc)
+
+    def withRemoteApiEndpoint(endpoint: RemoteApiEndpoint) =
+      copy(remoteApiEndpoint = Some(endpoint))
+  }
+
+  final case class RemoteApiEndpoint(
+      host: String,
+      port: Integer,
+      tlsConfig: Option[TlsConfiguration]) {
+    def withHost(host: String) = copy(host = host)
+    def withPort(port: Int) = copy(port = port)
+    def withTlsConfig(tlsConfig: TlsConfiguration) = copy(tlsConfig = Some(tlsConfig))
+    def withTlsConfigOption(tlsConfig: Option[TlsConfiguration]) = copy(tlsConfig = tlsConfig)
```

Comment:
why do we need this? shouldn't this be `withNoTlsConfig` instead? My point is that having both in this form seems redundant as we could keep only the optional.